File: Data Structures & Algorithms/is-anagram/submission-4.cpp
#include <array>

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size())
            return false;

        std::array<int, 26> dict {0};

        for (std::size_t i = 0; i < s.size(); ++i) {
            ++dict[s[i] - 'a'];
            --dict[t[i] - 'a'];
        }

        for (auto freq : dict)
            if (freq != 0)
                return false;

        return true;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        answer = []

        key_freq = {}

        for num in nums:
            key_freq[num] = 1 + key_freq.get(num, 0)
        
        buckets = [[] for _ in range(len(nums) + 1)]

        for key, freq in key_freq.items():
            buckets[freq].append(key)

        for bucket in reversed(buckets):
            if len(bucket) == 0:
                continue
            else:
                for key in bucket:
                    if k > 0:
                        k -= 1
                        answer.append(key)
                    else:
                        return answer 

        return answer